app/controllers/course/survey/surveys_controller.rb

```diff
@@ -6,7 +6,9 @@ class Course::Survey::SurveysController < Course::ComponentController
   def index
     respond_to do |format|
       format.html
-      format.json
+      format.json do
+        @surveys = @surveys.includes(:responses)
```

Comment:
It might be even slower when there are lots of students in the course. We should only load the responses of the current student. 

Do you want to make a separate PR for this ?